t/myriad.t

```diff
@@ -2,18 +2,138 @@ use strict;
+my $command = 'test';
+my $command_is_called = 0;
+$command_module->mock($command, async sub { my ($self, $param) = @_; $command_is_called = $param; });
 
 my $myriad = new_ok('Myriad');
-isa_ok(my $f = $myriad->shutdown_future, 'Future');
-is(refaddr($f), refaddr($myriad->shutdown_future), 'same Future on multiple calls');
-is(exception {
-    $myriad->shutdown->get
-}, undef, 'can shut down without exceptions arising');
-is($f->state, 'done', 'shutdown future marked as done');
+my $metaclass = $myriad->META;
+
+subtest "class methods and proper initialization" => sub {
+    can_ok($myriad, $_) for qw(configure_from_argv loop registry redis rpc_client rpc http subscription storage add_service service_by_name ryu shutdown run);
+
+
+    my $command_param = 'Testing';
+    $myriad->configure_from_argv(('-l', 'debug', '--subscription_transport', 'perl', '--rpc_transport', 'perl', '--storage_transport', 'perl', $command, $command_param));
+
+    # Check configure_from_argv init objects
+    my $loop = $metaclass->get_slot('$loop')->value($myriad);
+    isa_ok($loop, 'IO::Async::Loop::Poll', 'Loop is set');
+    isa_ok($myriad->config, 'Myriad::Config', 'Config is set');
+
+    # Logging setup
+    is($myriad->config->log_level, 'debug', 'Log level matching');
+    isa_ok(@{$myriad->config->log_level->{subscriptions}}[0], 'CODE', 'Logging has been setup');
+
+    # Tracing setup
+    isa_ok($metaclass->get_slot('$tracing')->value($myriad), 'Net::Async::OpenTracing', 'Tracing is set');
+    my $shutdown_tasks = $metaclass->get_slot('$shutdown_tasks')->value($myriad);
+    isa_ok($shutdown_tasks->[-1], 'CODE', 'Added to shutdown tasks');
+    is(@$shutdown_tasks, 1, 'One added shutdown task');
+
+    my $current_notifiers = loop_notifiers($myriad->loop);
+    ok($current_notifiers->{'Net::Async::OpenTracing'}, 'Tracing is added to  loop');
+
+    # Redis setup
+    isa_ok($metaclass->get_slot('$redis')->value($myriad), 'Myriad::Transport::Redis', 'Redis is set');
+    ok($current_notifiers->{'Myriad::Transport::Redis'}, 'Redis is added to  loop');
+
+    # Command
+    isa_ok($metaclass->get_slot('$commands')->value($myriad), 'Myriad::Commands', 'Command is set');
+    like($command_is_called, qr/$command_param/, 'Test Command has been found and called');
+
+};
+
+subtest "Myriad attributes setting tests" => sub {
+
+    # RPC
+    my $rpc = $myriad->rpc;
+    isa_ok($metaclass->get_slot('$rpc')->value($myriad), 'Myriad::RPC::Implementation::Perl', 'Myriad RPC is set');
+    my $current_notifiers = loop_notifiers($myriad->loop);
+    ok($current_notifiers->{'Myriad::RPC::Implementation::Perl'}, 'RPC is added to loop');
+    my $shutdown_tasks = $metaclass->get_slot('$shutdown_tasks')->value($myriad);
+    isa_ok($shutdown_tasks->[-1], 'CODE', 'Added to shutdown tasks');
+    is(@$shutdown_tasks, 2, 'Two added shutdown tasks');
+
+    # RPC Client
+    TODO: {
+        local $TODO = "Make perl implementation for RPC Client";
```

Comment:
Maybe a [`todo_skip`](https://metacpan.org/pod/Test::More#todo_skip) instead so you don't have to comment below.